docs/whatsnew/3.1.rst

```diff
@@ -10,6 +10,7 @@ Bug fixes
   :meth:`.Group.create_virtual_dataset` (:issue:`1732`).
 * Fix writing data as a list to a dataset with a sub-array data type
   (:issue:`1735`).
+* Fix exception handling in :meth:`.File.visititems` (:issue:`1740`)
```

Comment:
3.1 was already released - your changes will be in 3.2. To add a release note, make a new file in the `news/` folder by copying `TEMPLATE.rst`. The files in this directory will be pulled together when we do a release.

More about this in the contribution docs: https://docs.h5py.org/en/stable/contributing.html#write-a-release-note